src/realm/object-store/c_api/sync.cpp

```diff
@@ -410,6 +411,66 @@ RLM_API void realm_sync_config_set_resync_mode(realm_sync_config_t* config,
     config->client_resync_mode = ClientResyncMode(mode);
 }
 
+RLM_API realm_object_id_t realm_flx_sync_subscription_id(const realm_flx_sync_subscription_t* subscription) noexcept
+{
+    REALM_ASSERT(subscription != nullptr);
+    return wrap_err([&]() {
+        return to_capi(subscription->id());
+    });
+}
+
+RLM_API bool realm_flx_sync_subscription_has_name(const realm_flx_sync_subscription_t* subscription) noexcept
+{
+    REALM_ASSERT(subscription != nullptr);
+    return wrap_err([&]() {
+        return subscription->has_name();
+    });
+}
+
+RLM_API const char* realm_flx_sync_subscription_name(const realm_flx_sync_subscription_t* subscription) noexcept
```

Comment:
Just a small note: Cpp `string_views` are not supposed to hold a reference to a `'\0'`. So if for any reason core changes the container from `std::string` to something else that does not guarantee that the strings will be terminated correctly, this code could lead to crashes. 
However, allocating a new string every time seems too expensive.

Reply:
Yes. We will keep using `.data()` directly here, since it is backed by a `std::string`. We could probably change these to `const std::String` and expose directly on `Subscription` class, but I will leave that to others

Reply:
We have `realm_string_t`, which is the C API equivalent of `StringData`, and it already means "string buffer owned by someone else". Right now it's only used for strings actually coming from the database, but I guess they would fit here as well. I'll be happy to change this in a follow-up PR if you think it's a good idea, @nicola-cab.

Reply:
Yes, personally, I would do that. Better don't rely on the fact that `std::string` is null terminated, since string_view is just a view of the characters.

Reply:
Feel free to commit on this branch